aries_vcx/src/handlers/connection/invitee/state_machine.rs

```diff
@@ -246,106 +244,88 @@ impl SmConnectionInvitee {
     pub fn handle_invitation(self, invitation: Invitation) -> VcxResult<Self> {
         let Self { state, .. } = self;
         let state = match state {
-            InviteeFullState::Initial(state) => InviteeFullState::Invited((state, invitation).into()),
+            InviteeFullState::Initial(state) => InviteeFullState::Invited((state, invitation.clone()).into()),
             _ => state.clone()
         };
-        Ok(Self { state, ..self })
+        Ok(Self { state, thread_id: invitation.get_id()?, ..self })
     }
 
     pub fn handle_connect(self, routing_keys: Vec<String>, service_endpoint: String) -> VcxResult<Self> {
-        let Self { source_id, pairwise_info, state, send_message } = self;
-        let state = match state {
+        let state = match self.state {
             InviteeFullState::Invited(state) => {
-                let recipient_keys = vec!(pairwise_info.pw_vk.clone());
+                let recipient_keys = vec!(self.pairwise_info.pw_vk.clone());
                 let request = Request::create()
-                    .set_label(source_id.to_string())
-                    .set_did(pairwise_info.pw_did.to_string())
+                    .set_label(self.source_id.to_string())
+                    .set_did(self.pairwise_info.pw_did.to_string())
                     .set_service_endpoint(service_endpoint)
+                    .set_thread_id(&self.thread_id)
                     .set_keys(recipient_keys, routing_keys);
 
                 let ddo = DidDoc::from(state.invitation.clone());
-                send_message(&pairwise_info.pw_vk, &ddo, &request.to_a2a_message())?;
+                (self.send_message)(&self.pairwise_info.pw_vk, &ddo, &request.to_a2a_message())?;
                 InviteeFullState::Requested((state, request).into())
             }
-            _ => {
-                state.clone()
-            }
+            _ => self.state.clone()
         };
-        Ok(Self { state, pairwise_info, source_id, ..self })
+        Ok(Self { state, ..self })
     }
 
     pub fn handle_connection_response(self, response: SignedResponse) -> VcxResult<Self> {
-        verify_thread_id(&self.get_thread_id()?, &A2AMessage::ConnectionResponse(response.clone()))?;
```

Comment:
I think we should be doing this check, why remove?

Reply:
We are doing them multiple times - once in `update_state` and once in the handlers I removed the from.

Reply:
Oh yeah! I missed that. However I think it should actually be in `state_machine.rs` - even if it means that it needs to be placed at multiple `handle_msg_<xyz>` functions. The reason being - `thread_id` is owned by the connection state machine itself and should happen regardless of how the upper abstraction layer (`connection.rs`, or perhaps it's agency-less equivalent in future) is implemented.